### DISCUSSION TITLE
Clangd fails to work with "unusual" setups

### DISCUSSION THREAD

**Discussion by HFDan** [clangd/clangd#1039] [Q&A] [answered]
My setup consists of running msvc under wine. The problem with this is that clangd is unable to detect anything related to include paths, including the STL.

I understand that this might not be an expected use-case, but getting clangd to work with this setup would be great!

**Comment by HighCommander4** [comment]
While I haven't worked with this specific setup, other "unusual" setups that have come up can typically be made to work using the techniques described in [this troubleshooting entry](https://clangd.llvm.org/troubleshooting#cant-find-standard-library-headers-map-stdioh-etc).

If you tried that and it hasn't worked, please share some more details (e.g. clangd logs, your `compile_commands.json` and `.clangd` files if applicable, where the compiler / stdlib is installed on your system).

**Reply by kadircet** [reply]
In addition to what @HighCommander4 has mentioned, it would be also great to hear if you're running clangd in your local system or through wine as well. It's quite likely that you only have your libraries visible to wine environment and clangd is unable to pick them up because they're in "unusual" places that clang cannot detect when running inside your host machine.

**Comment by HFDan** [comment]
> While I haven't worked with this specific setup, other "unusual" setups that have come up can typically be made to work using the techniques described in [this troubleshooting entry](https://clangd.llvm.org/troubleshooting#cant-find-standard-library-headers-map-stdioh-etc).
> 
> If you tried that and it hasn't worked, please share some more details (e.g. clangd logs, your `compile_commands.json` and `.clangd` files if applicable, where the compiler / stdlib is installed on your system).


--query-driver does not work, since it gives gcc arguments to the compiler, when the compiler expects msvc arguments

**Reply by HighCommander4** [reply]
Are you able to compile your project with `clang-cl`? My understanding is `clang-cl` is supposed to be a drop-in replacement for msvc, including the command-line syntax.

**Reply by HFDan** [reply]
clang-cl does not create windows executables. It creates linux executables. It's there so you do not have to adapt cl commands to gcc.

**Reply by HighCommander4** [reply]
When running Windows natively, you can definitely use `clang-cl` to produce a Windows executable. I imagine it should be able to do the same when run under wine, but I haven't tried. (But it would need to run under wine, not natively on Linux, that's true.)

**Reply by HFDan** [reply]
I am unable to compile the program using clang-cl. the msvc in wine setup works just fine though.

**Reply by HFDan** [reply]
Eventually i managed to get clang-cl to work with the following args:
`/winsysroot /opt/msvc/`, and symlinking some directories in /opt/msvc

**Comment by HFDan** [comment, marked as answer]
Solution:
1. Go to `compile_commands.json`
2. Use `sed` to replace all occurences of `/opt/msvc/bin/x64/cl` with `clang-cl /winsysroot /opt/msvc`.
3. Profit
